### PR TITLE
Bigtable: Converging return types of row.commit

### PR DESCRIPTION
**Pull request by IlyaFaer**
https://github.com/googleapis/google-cloud-python/issues/8451

### CODE REVIEW

File: bigtable/google/cloud/bigtable/row.py [hunk 1]
```diff
@@ -82,6 +82,21 @@ def table(self):
...
+        For example:
+
+        .. literalinclude:: snippets_table.py
+            :start-after: [START bigtable_row_row_content]
+            :end-before: [END bigtable_row_row_content]
+
+        :rtype: row_content: :class:`PartialRowData <google.cloud.bigtable.row_data.PartialRowData>`
+        :returns: Contents of the row read from table.
+        """
+        return self.table.read_row(self.row_key)
```
Comment (IlyaFaer) [inline bigtable/google/cloud/bigtable/row.py]: Added property in superclass, that mirrors to `read_row()`

**Comment by sduskis** [comment]
I think this PR was premature. The issue was raised without understanding why things are the way they are.

File: bigtable/google/cloud/bigtable/row.py [hunk 2]
```diff
@@ -451,11 +466,15 @@ def commit(self):
...
 
+        :rtype: row_content: :class:`PartialRowData <google.cloud.bigtable.row_data.PartialRowData>`
+        :returns: Contents of the row.
+
         :raises: :exc:`~.table.TooManyMutationsError` if the number of
                  mutations is greater than 100,000.
         """
         self._table.mutate_rows([self])
         self.clear()
+        return self.row_content
```
Comment (sduskis) [inline bigtable/google/cloud/bigtable/row.py]: This is really bad.  Its an extra service call for no reason.

File: bigtable/google/cloud/bigtable/row.py [hunk 3]
```diff
@@ -578,15 +597,15 @@ def commit(self):
...
+        data_client.check_and_mutate_row(
             table_name=self._table.name,
             row_key=self._row_key,
             predicate_filter=self._filter.to_pb(),
             true_mutations=true_mutations,
             false_mutations=false_mutations,
         )
         self.clear()
-        return resp.predicate_matched
+        return self.row_content
```
Comment (sduskis) [inline bigtable/google/cloud/bigtable/row.py]: This changes the existing behavior, and doesn't match the service's results.  This actually will reduce usage and performance.

File: bigtable/google/cloud/bigtable/row.py [hunk 4]
```diff
@@ -907,15 +923,13 @@ def commit(self):
...
+        data_client.read_modify_write_row(
             table_name=self._table.name, row_key=self._row_key, rules=self._rule_pb_list
         )
 
         # Reset modifications after commit-ing request.
         self.clear()
-
-        # NOTE: We expect row_response.key == self._row_key but don't check.
-        return _parse_rmw_row_response(row_response)
+        return self.row_content
```
Comment (sduskis) [inline bigtable/google/cloud/bigtable/row.py]: This is also bad.  This causes an extra lookup.  `row_response` is already equivalent to `row_content` without requiring a lookup

Comment (IlyaFaer) [reply bigtable/google/cloud/bigtable/row.py]: As I understood, the request was to return the whole row, while `row_content` contains only modified cells
https://github.com/googleapis/google-cloud-python/blob/fa6e4f5784f0063ff832159e561aa89d45497dea/bigtable/google/cloud/bigtable/row.py#L939

File: bigtable/google/cloud/bigtable/row.py (hunk 3 above)
Comment (IlyaFaer) [reply bigtable/google/cloud/bigtable/row.py]: I knew, it's an extra call, but reading row by indexed key looks lightly (if number/size of columns is not too large, of course). And changing the behavior was requested in issue. Is there are any indicator to understand, that issue approved by people, who know the details? It's not first time, when I PR'ed too fast. For newbie like me some things are not obvious (especially, when googlers said that it's not bad idea)

**Pull request closed without merging**